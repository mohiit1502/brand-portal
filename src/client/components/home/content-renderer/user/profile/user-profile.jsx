import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Http from "../../../../../utility/Http";
import {updateUserProfile} from "../../../../../actions/user/user-actions";
import {dispatchDiscardChanges, TOGGLE_ACTIONS, toggleModal} from "../../../../../actions/modal-actions"
import {showNotification} from "../../../../../actions/notification/notification-actions"
import InputFormatter from "../../../../../utility/phoneOps";
import Helper from "../../../../../utility/helper";
import ContentRenderer from "../../../../../utility/ContentRenderer";
import Validator from "../../../../../utility/validationUtil";
// import FORMFIELDCONFIG from "./../../../../../config/formsConfig/form-field-meta";
import "../../../../../styles/home/content-renderer/user/profile/user-profile.scss";
import CONSTANTS from "../../../../../constants/constants";
import mixpanel from "../../../../../utility/mixpanelutils";
import MIXPANEL_CONSTANTS from "../../../../../constants/mixpanelConstants";
import Cookies from "electrode-cookies";

class UserProfile extends React.Component {

  constructor (props) {
    super(props);
    const functions = ["bubbleValue", "displayChangePassword", "displayManageProfileNotification", "isDirty", "onChange", "setFormData", "disableInput", "saveUser"];
    functions.forEach(name => this[name] = this[name].bind(this));
    this.validateState = Validator.validateState.bind(this);
    this.getFieldRenders = ContentRenderer.getFieldRenders.bind(this);
    this.loader = Helper.loader.bind(this);
    this.onInvalid = Validator.onInvalid.bind(this);
    this.invalid = {firstName: false, lastName: false, companyName: false, phone: false};
    const userProfileConfiguration = this.props.userProfileContent ? this.props.userProfileContent : {}

    this.state = {
      section: {...userProfileConfiguration.sectionConfig},
      form: {
        ...userProfileConfiguration.formConfig,
        inputData: {...userProfileConfiguration.fields},
        // underwritingChecked: false,
      },
      isSeller: Cookies.get("client_type") === "seller"
    };

    const formatter = new InputFormatter();
    const handlers = formatter.on(`#${this.state.section.id}-${this.state.form.inputData.phone.inputId}-custom-input`);
    this.prebounceChangeHandler = handlers.inputHandler;

    Object.keys(this.state.form.inputData).forEach(itemKey => {
      const item = this.state.form.inputData[itemKey];
      this.state.form.inputData[itemKey].value = Helper.search(item.initValuePath, this.props.userProfile);
      if  (itemKey === "phone") {
        const value = this.state.form.inputData[itemKey].value;
        this.state.form.inputData[itemKey].value = (value === "0000000000") || (value === "(000) 000-0000") ? "" : value;
      }
    });
  }

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      this.setFormData(this.props.userProfile);
    }
  }

  isDirty () {
    const {firstName: originalFName, lastName: originalLName, email: originalEmail, phoneNumber: originalPhone, companyName: originalCompany} = this.props.userProfile;
    const {firstName: {value: currentFName}, lastName: {value: currentLName}, emailId: {value: currentEmail}, phone: {value: currentPhone}} = this.state.form.inputData;
    const currentCompany = this.state.form.inputData.company ? this.state.form.inputData.company.value : "";
    const originalPhoneModified = (originalPhone === "0000000000" || originalPhone === "(000) 000-0000") ? "" : originalPhone;
    return originalFName !== currentFName || originalLName !== currentLName || originalEmail !== currentEmail || originalPhoneModified !== currentPhone || (originalCompany && currentCompany && originalCompany !== currentCompany);
      // || ((!originalCompany && currentCompany) || (originalCompany && !currentCompany) || (originalCompany !== currentCompany));
  }

  setFormData(obj) {
    if (this.props.shouldDiscard) {
      const form = {...this.state.form};
      form.isDisabled = true;
      form.inputData.firstName.value = obj.firstName;
      form.inputData.lastName.value = obj.lastName;
      form.inputData.companyName.value = obj.type === "ThirdParty" ? obj.companyName : "";
      form.inputData.emailId.value = obj.email;
      form.inputData.phone.value = (obj.phoneNumber === "0000000000") || (obj.phoneNumber === "(000) 000-0000") ? "" : obj.phoneNumber;

      Object.keys(form.inputData).forEach(key => {
        if (key !== "emailId") {
          const item = form.inputData[key];
          item.disabled = true;
          item.error = "";
        }
      });

      this.setState({form});
      this.props.dispatchDiscardChanges(false);
    }
  }

  bubbleValue (evt, key, error) {
    const targetVal = evt.target.value;
    this.setState(state => {
      state = {...state};
      state.form.inputData[key].value = targetVal;
      state.form.inputData[key].error = error;
      return state;
    });
  }

  onChange (evt, key) {
    if (evt && evt.target) {
      if (evt.target.pattern && evt.target.checkValidity()) {
        this.invalid[key] = false;
      }
      const targetVal = evt.target.value;
      this.setState(state => {
        state = {...state};
        const inputData = state.form.inputData;
        state.form.inputData[key].value = targetVal;
        inputData[key].error = !this.invalid[key] ? "" : inputData[key].error;
        this.invalid[key] = false;
        return {
          ...state
        };
      });
    }
  }

  displayManageProfileNotification () {
    let template = this.props.modalsMeta && this.props.modalsMeta.PASSWORD_RESET_SELLER;
    template = {templateName: "StatusModalTemplate", image: "", ...template};
    this.props.toggleModal(TOGGLE_ACTIONS.SHOW, {...template});
    mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.CHANGE_PASSWORD.DISPLAY_SELLER_MANAGE_PROFILE, {WORK_FLOW: "EDIT_USER_PROFILE"});
  }

  displayChangePassword() {
    const meta = { templateName: "ResetPasswordTemplate" };
    this.props.toggleModal(TOGGLE_ACTIONS.SHOW, {...meta});
    mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.CHANGE_PASSWORD.DISPLAY_CHANGE_PASSWORD, {WORK_FLOW: "EDIT_USER_PROFILE"});
  }

  disableInput (disable) {
    disable = !!disable;
    if (disable && this.isDirty()) {
      const meta = { templateName: "Alert" };
      this.props.toggleModal(TOGGLE_ACTIONS.SHOW, {...meta});
      mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.EDIT_USER_PROFILE.CANCEL_EDIT_PROFILE, {WORK_FLOW: "EDIT_USER_PROFILE"});
    } else {
      const form = {...this.state.form};
      form.isDisabled = disable;
      form.inputData.phone.disabled = disable;
      Object.keys(form.inputData).forEach(key => {
        if (key !== "emailId") {
          const item = form.inputData[key];
          item.disabled = disable;
          item.error = "";
        }
      });
      this.setState({form});
      if (!disable) mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.EDIT_USER_PROFILE.EDIT_PROFILE, {WORK_FLOW: "EDIT_USER_PROFILE"});
    }
  }

  // toggleUnderwritingCheck () {
  //   const form = {...this.state.form};
  //   form.underwritingChecked = !form.underwritingChecked;
  //   this.setState({form});
  // }

  async saveUser (evt) {
    evt.preventDefault();

    if (!this.validateState()) {
      this.loader("form", true);
      const loginId = this.state.form.inputData.emailId.value;
      const firstName = this.state.form.inputData.firstName.value;
      const lastName = this.state.form.inputData.lastName.value;
      const phoneNumber = this.state.form.inputData.phone.value ? this.state.form.inputData.phone.value : "0000000000"; //[note:to handle VIP phone number validation]
      const payload = {
        user: {
          loginId,
          firstName,
          lastName,
          phoneNumber,
          properties: {
            companyName: this.state.form.inputData.companyName.value
          }
        }
      };

      const url = this.state.form.apiPath;
      if (this.isDirty()) {
        const mixpanelPayload = {
          API: url,
          WORK_FLOW: "EDIT_USER_PROFILE"
        };
        return Http.put(`${url}/${payload.user.loginId}`, payload, null, null, this.props.showNotification, this.state.form.profileSaveMessage)
          .then(async res => {
            this.loader("form", false);
            this.props.updateUserProfile(res.body);
            this.disableInput(true);
            mixpanelPayload.API_SUCCESS = true;
          })
          .catch(err => {
            this.loader("form", false);
            mixpanelPayload.API_SUCCESS = false;
            mixpanelPayload.ERROR = err.message ? err.message : err;
          })
          .finally(() => {
            mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.EDIT_USER_PROFILE.SAVE_PROFILE, mixpanelPayload);
          });
      } else {
        this.loader("form", false);
        this.disableInput(true);
      }
    }
    return null;
  }

  render () {
    return (
      <div className={`user-profile-content h-100${this.state.form.loader ? " loader" : ""}`}>
        <header className={this.state.section.headerClasses}>{this.state.section.sectionTitle}</header>
        <form className={this.state.section.formClasses} autoComplete="off" onSubmit={this.saveUser} style={{marginRight: "50%"}}>
          {this.getFieldRenders()}
        </form>
      </div>
    );
  }
}

UserProfile.propTypes = {
  dispatchDiscardChanges: PropTypes.func,
  shouldDiscard: PropTypes.bool,
  showNotification: PropTypes.func,
  toggleModal: PropTypes.func,
  userProfile: PropTypes.object,
  updateUserProfile: PropTypes.func
};

const mapStateToProps = state => {
  return {
    modalsMeta: state.content.metadata ? state.content.metadata.MODALSCONFIG : {},
    userProfileContent: state.content && state.content.metadata && state.content.metadata.SECTIONSCONFIG && state.content.metadata.SECTIONSCONFIG.USERPROFILE,
    shouldDiscard: state.modal.shouldDiscard,
    userProfile: state.user.profile
  };
};

const mapDispatchToProps = {
  dispatchDiscardChanges,
  showNotification,
  toggleModal,
  updateUserProfile
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserProfile);
