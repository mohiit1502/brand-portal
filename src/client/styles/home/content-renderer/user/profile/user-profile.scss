@import "../../../../override/variables";

.user-profile-content {
  .content-header-row {
    font-weight: bold;
  }
  .password-reset-col{
    button{
      border-radius: 42px;
      font-weight: bold;
      font-size: 15px;

      &.reset-password {
        border: $default-input-border;
        color: $market-place-blue;
      }

      &.manage-profile {
        color: white;
      }

      &:hover {
        color: white;
        background: $market-place-blue;
      }
    }
  }

  .edit-button-panel {
    .btn-primary{
      border-radius: 42px;
      background: #006799;
      font-weight: bold;
      font-size: 15px;
      border: none;
    }
    .btn-link{
      font-weight: bold;
      font-size: 18px;
      text-decoration: underline;
    }
  }
  .password-reset-row {
    .badge {
      //border: 1px solid $blue;
    }
  }
}
