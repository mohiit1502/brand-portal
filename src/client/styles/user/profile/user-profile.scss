@import "../../override/variables";

.user-profile-content {
  .password-reset-row {
    .badge {
      border: 1px solid $blue;
    }
  }
}
