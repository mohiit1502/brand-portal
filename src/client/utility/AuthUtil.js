import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
