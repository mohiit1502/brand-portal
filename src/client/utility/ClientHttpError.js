export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
