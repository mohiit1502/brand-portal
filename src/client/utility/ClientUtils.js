import queryString from "query-string";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }
}
