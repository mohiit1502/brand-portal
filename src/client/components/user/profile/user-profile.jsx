import React from "react";
import { connect } from "react-redux";
import "../../../styles/user/profile/user-profile.scss";
import PropTypes from "prop-types";
import CustomInput from "../../custom-input/custom-input";

class UserProfile extends React.Component {

  constructor (props) {
    super(props);
    this.onInputChange = this.onInputChange.bind(this);
    this.setFormData = this.setFormData.bind(this);
    this.disableInput = this.disableInput.bind(this);
    this.toggleUnderwritingCheck = this.toggleUnderwritingCheck.bind(this);

    this.state = {
      form: {
        isDisabled: true,
        underwritingChecked: false,
        id: "user-profile-form",
        inputData: {
          firstName: {
            label: "First Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          lastName: {
            label: "Last Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          companyName: {
            label: "Company Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          emailId: {
            label: "Email ID",
            required: true,
            value: "",
            type: "email",
            pattern: null,
            disabled: true
          },
          phone: {
            label: "Mobile Number",
            required: false,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          }
        }
      }
    };
  }

  componentDidMount() {
    const obj = {
      firstName: "Shubhansh",
      lastName: "Sahai",
      companyName: "Walmart Labs",
      emailId: "shubhansh.sahai@walmartlabs.com",
      phone: "9686648597"
    };

    this.setFormData(obj);

  }

  setFormData(obj) {
    const form = {...this.state.form};
    form.inputData.firstName.value = obj.firstName;
    form.inputData.firstName.disabled = true;
    form.inputData.lastName.value = obj.lastName;
    form.inputData.lastName.disabled = true;
    form.inputData.companyName.value = obj.companyName;
    form.inputData.companyName.disabled = true;
    form.inputData.emailId.value = obj.emailId;
    form.inputData.emailId.disabled = true;
    form.inputData.phone.value = obj.phone;
    form.inputData.phone.disabled = true;

    this.setState({
      form
    });
  }

  onInputChange (evt, key) {
    if (evt && evt.target) {
      const targetVal = evt.target.value;
      this.setState(state => {
        state = {...state};
        state.form.inputData[key].value = targetVal;
        return {
          ...state
        };
      });
    }
  }

  disableInput (disable) {
    disable = !!disable;

    const form = {...this.state.form};

    form.isDisabled = disable;

    form.inputData.firstName.disabled = disable;
    form.inputData.lastName.disabled = disable;
    form.inputData.companyName.disabled = disable;
    form.inputData.emailId.disabled = disable;
    form.inputData.phone.disabled = disable;

    this.setState({
      form
    });

  }

  toggleUnderwritingCheck () {
    const form = {...this.state.form};
    form.underwritingChecked = !form.underwritingChecked;
    this.setState({form});
  }

  render () {

    return (
      <div className="row user-profile-content h-100">
        <div className="col h-100">
          <div className="row h-10">
            <div className="col">
              <h3>User Profile</h3>
            </div>
          </div>
          <div className="row mt-4 h-90">
            <div className="col">
              <form className="h-100" autoComplete="off">
                <div className="row h-60">
                  <div className="col h-100">
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"firstName"}
                          inputId={"firstName"}
                          formId={this.state.form.id} label={this.state.form.inputData.firstName.label}
                          required={this.state.form.inputData.firstName.required} value={this.state.form.inputData.firstName.value}
                          type={this.state.form.inputData.firstName.type} pattern={this.state.form.inputData.firstName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.firstName.disabled} />
                      </div>
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"lastName"}
                          inputId={"lastName"}
                          formId={this.state.form.id} label={this.state.form.inputData.lastName.label}
                          required={this.state.form.inputData.lastName.required} value={this.state.form.inputData.lastName.value}
                          type={this.state.form.inputData.lastName.type} pattern={this.state.form.inputData.lastName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.lastName.disabled} />
                      </div>
                    </div>
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"companyName"}
                          inputId={"companyName"}
                          formId={this.state.form.id} label={this.state.form.inputData.companyName.label}
                          required={this.state.form.inputData.companyName.required} value={this.state.form.inputData.companyName.value}
                          type={this.state.form.inputData.companyName.type} pattern={this.state.form.inputData.companyName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.companyName.disabled} />
                      </div>
                    </div>
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"emailId"}
                          inputId={"emailId"}
                          formId={this.state.form.id} label={this.state.form.inputData.emailId.label}
                          required={this.state.form.inputData.emailId.required} value={this.state.form.inputData.emailId.value}
                          type={this.state.form.inputData.emailId.type} pattern={this.state.form.inputData.emailId.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.emailId.disabled} />
                      </div>
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"phone"}
                          inputId={"phone"}
                          formId={this.state.form.id} label={this.state.form.inputData.phone.label}
                          required={this.state.form.inputData.phone.required} value={this.state.form.inputData.phone.value}
                          type={this.state.form.inputData.phone.type} pattern={this.state.form.inputData.phone.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.phone.disabled} />
                      </div>
                    </div>
                    <div className="row mb-5 password-reset-row">
                      <div className="col-3">
                        <div className="btn btn-primary btn-sm no-border-radius px-3">
                          Password Reset
                        </div>
                        <span className="badge badge-pill bg-white text-blue ml-2 cursor-pointer"> ? </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div className="row h-40">
                  <div className="col h-100">
                    <div className="row pt-5">
                      <div className="col-xl-6 col-10">
                        <label>
                          <input type="checkbox" className="mr-2" checked={this.state.form.underwritingChecked} onChange={this.toggleUnderwritingCheck}/>
                          I have a good faith belief that the use of the material in the manner complained of is not authorized by the copyright owner, its agent, or the law.
                        </label>
                      </div>
                    </div>
                    <div className="row pt-5">
                      <div className="col-xl-3 col-5" />
                      <div className="col-xl-3 col-5 text-right">
                        {
                          this.state.form.isDisabled ?
                            (
                              <div className="btn btn-primary btn-sm no-border-radius px-4" onClick={() => this.disableInput(false)}>
                                Edit
                              </div>
                            ) :
                            (
                              <span>
                                <div className="btn btn-link font-size-14 px-4 mr-3" onClick={() => this.disableInput(true)}>Cancel</div>
                                <button type="submit" className="btn btn-primary btn-sm no-border-radius px-4 font-size-14" disabled={!this.state.form.underwritingChecked}>
                                  Save
                                </button>
                              </span>
                            )
                        }

                      </div>
                    </div>
                  </div>
                </div>

              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => state;

export default connect(
  mapStateToProps,
  dispatch => ({dispatch})
)(UserProfile);
