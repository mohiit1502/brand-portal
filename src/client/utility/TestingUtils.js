import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const realStore = state => {
  const configureStore = initialState => {
    const composeEnhancers =
      typeof window === "object" &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          serialize: { // prettier-ignore
            immutable: Immutable
          }
        }) : compose;
    const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunk)));

    if (module.hot) {
      module.hot.accept("./reducers", () => {
        const nextRootReducer = require("./reducers").default;
        store.replaceReducer(nextRootReducer);
      });
    }

    return store;
  };

  return configureStore(state || window.__WML_REDUX_INITIAL_STATE__);
}

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
