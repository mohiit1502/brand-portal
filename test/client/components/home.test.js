import React, {useRef} from "react";
import ReactDOM from "react-dom";
import Home from "../../../src/client/components/home/home";
import 'regenerator-runtime';
import {testStore} from "../../../src/client/utility/TestingUtils";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
// import FORMFIELDCONFIG from "../../../src/client/config/formsConfig/form-field-meta";
import FORMFIELDMOCK from "../mocks/formFeildMock";
import MODALSMETA from "../../../src/client/config/contentDescriptors/modals-meta"
import homePageMocks from "../mocks/homePageMocks";
import profile from "../mocks/userProfile";
import portalRegUserProfile from "../mocks/portalRegUserProfile";


let store;
const setUp = (props) => {
  const mockStore = {
    modal:{
      enable:false,
      template:null
    },
    content:{
      metadata: {...FORMFIELDMOCK, ...MODALSMETA}
    },
    user: {profile}
  };
  store = testStore(mockStore);
  return renderer.create(<Provider store={store}><BrowserRouter><Home {...props} /></BrowserRouter></Provider>);
};

const setUp2 = (props) => {
  const mockStore = {
    modal:{
      enable:false,
      template:null
    },
    content:{
      metadata: {...FORMFIELDMOCK, ...MODALSMETA}
    },
    user: {portalRegUserProfile}
  };
  store = testStore(mockStore);
  return renderer.create(<Provider store={store}><BrowserRouter><Home {...props} /></BrowserRouter></Provider>);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});

describe("Home Tests", () => {

  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });

  it("renders without error", () => {
    const mRef = {current: document.createElement("div")};
    useRef.mockReturnValue(mRef);
    wrapper = setUp({homePageMocks});
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it("renders onboarding screen without error", () => {
  //   const mRef = {current: document.createElement("div")};
  //   useRef.mockReturnValue(mRef);
  //   wrapper = setUp2({
  //     ...homePageMocks,
  //     userProfile:portalRegUserProfile
  //   });
  //   const tree = wrapper.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});




describe("Home", () => {
  let component;
  let container;

  beforeEach(() => {
    container = document.createElement("div");
  });

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(container);
  });

  it("has expected content with deep render", () => {
    const initialState = {
      checkBox: { checked: false },
      number: { value: 999 }
    };
    expect(true).toBe(true);

    // const store = createStore(rootReducer, initialState);
    //
    // component = ReactDOM.render(
    //   <Provider store={store}>
    //     <BrowserRouter>
    //       <Home />
    //     </BrowserRouter>
    //   </Provider>,
    //   container
    // );
    //
    // expect(component).to.not.be.false;
  });
});
