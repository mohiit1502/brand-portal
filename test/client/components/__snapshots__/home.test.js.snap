// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Tests renders without error 1`] = `
<div
  className="view-container home-container"
>
  <nav
    className="navbar navbar-expand-md navbar-dark home-header-nav home-header-nav-z-index"
  >
    <a
      className="navbar-brand walmart-brand"
      href="/dashboard"
      onClick={[Function]}
    >
      <img
        src="WMWhite-horizontal.svg"
      />
    </a>
    <button
      aria-controls="collapsible-header"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#collapsible-header"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse navbar-collapsible-header"
      id="collapsible-header"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item nav-item-help mx-4"
        >
          <a
            className="nav-link nav-help"
            href="/help"
            onClick={[Function]}
          >
            <img
              className="pr-2"
              height="32px"
              src="help-header.png"
            />
             Help
          </a>
        </li>
        <li
          className="nav-item dropdown nav-item-profile ml-4"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="nav-link user-name dropdown-toggle"
            data-toggle="dropdown"
            href="/"
            onClick={[Function]}
            role="button"
          >
            <img
              alt="Profile Pic"
              className="pr-2"
              height="32px"
              src="user-profile.png"
            />
            Test Mohit Nagar
          </a>
          <div
            className="dropdown-menu dropdown-menu-right no-border-radius shadow-sm mt-2"
          >
            <a
              className="dropdown-item"
              href="/profile"
              onClick={[Function]}
            >
              Profile
            </a>
            <a
              className="dropdown-item"
              href=""
              onClick={[Function]}
            >
              Logout
            </a>
          </div>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
